Makefile

```diff
@@ -250,16 +284,22 @@ endif
 
 
 .PHONY: test-e2e
+ifdef USE_PROMETHEUS
+test-e2e: PROMETHEUS_URL ?= http://localhost:9090
+endif
 ifdef USE_GOTESTSUM
 test-e2e: $(GOTESTSUM)
 endif
 test-e2e: TEST_ARGS ?=
 test-e2e: WHAT ?= ./test/e2e...
 test-e2e: build-all
-	UNSAFE_E2E_HACK_DISABLE_ETCD_FSYNC=true NO_GORUN=1 GOOS=$(OS) GOARCH=$(ARCH) \
+	PROMETHEUS_URL=$(PROMETHEUS_URL) UNSAFE_E2E_HACK_DISABLE_ETCD_FSYNC=true NO_GORUN=1 GOOS=$(OS) GOARCH=$(ARCH) \
```

Comment:
We might want to prepend the testing commands with `-` (and add a comment on why/what it does), since it's totally possible that the tests can fail. Based on my understanding, if you run `make start-prometheus test-e2e stop-prometheus` and `test-e2e` fails, `stop-prometheus` will not actually run. We still want the last 2 prometheus targets to run regardless of the test result.

[See the `make` docs about the `-` character](https://www.gnu.org/software/make/manual/make.html#Errors)

Reply:
good point 👍

Reply:
done

Reply:
[As discussed in Slack](https://kubernetes.slack.com/archives/C021U8WSAFK/p1676375034944059), move the failure handling logic into `hack` scripts instead of the Makefile.

Reply:
done